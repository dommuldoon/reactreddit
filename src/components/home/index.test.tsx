import Home from './';
import React, { useState } from 'react';
import { render, fireEvent, getByTestId, getByLabelText, getByDisplayValue, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

let store = mockStore({
	data: {
		items: { id: '1', name: 'subreddit' },
	},
});

describe('Home Component', () => {
	const comp = render(
		<Provider store={store}>
			<Home />
		</Provider>
	);

	it('renders', () => {
		const title = comp.getByText(/Subreddit/i);
		expect(title).toBeInTheDocument();
		expect(comp.container.firstChild).toMatchSnapshot();
	});

	it('should call onChange prop with value', async () => {
		const changeQuery = jest.fn();
		render(
			<input
				type="text"
				className="form-control"
				placeholder="subreddit"
				aria-label="subreddit"
				onChange={changeQuery}
			/>
		);
		const input: HTMLElement = comp.getByLabelText('subreddit');
		fireEvent.change(input, { target: { value: '23' } });
		expect(changeQuery).toBeCalled();
		expect(input.value).toBe('23');
	});
});
