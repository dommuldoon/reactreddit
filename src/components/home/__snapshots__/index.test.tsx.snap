// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders 1`] = `
<h1>
  Search Subreddits
</h1>
`;
